src/core/server/elasticsearch/client/cluster_client.test.ts

```diff
@@ -247,6 +268,31 @@ describe('ClusterClient', () => {
         headers: {
           foo: 'request',
           hello: 'dolly',
+          'x-opaque-id': expect.any(String),
+        },
+      });
+    });
+
+    it('respect the precedence of x-opaque-id header over config headers', () => {
```

Comment:
I'm actually not sure this is the desired behavior. Should operators be able to override this by setting a custom header via config? It seems very useless to do that, so I went with this behavior instead. The x-opaque-id header will be set by the system every time. Depending on the configuration, it will either be sourced from the incoming request or generated on the fly by the HttpService.

Reply:
@thomheymann I see your 👍 but not sure what your opinion here is? Do you think we should allow the user to override the `x-opaque-id` header via the `elasticsearch.customHeaders` config? I'm leaning towards yes, as I believe it would align more closely to the description of this config in our docs:

> Header names and values to send to Elasticsearch. Any custom headers cannot be overwritten by client-side headers, regardless of the `elasticsearch.requestHeadersWhitelist` configuration.

Reply:
Oh sorry, I actually agreed with your thinking that overriding that header with a static config value makes no sense really. 

However, I wasn't aware that that is literally the purpose of that config value so maybe for consistency it's best to follow that behaviour.

Reply:
If we follow the current behavior, we should allow `x-opaque-id` to be overwritten through `elasticsearch.customHeaders`. However, it seems allowing that here would break the linking for logging, correct? If so, I vote for keeping the current behavior of the PR and adding a notice to the `elasticsearch.customHeaders` property that `x-opaque-id` is not supported.

Reply:
> However, it seems allowing that here would break the linking for logging, correct?

Yes, it would break any traceability / auditability between Kibana requests and Elasticsearch requests. To me it seems like this is just a support case waiting to happen. I'm +1 on keeping the current behavior in the PR as well. I'll add a note to the docs for `elasticsearch.customHeaders` that the `X-Opaque-Id` header sent to ES cannot be configured with this option.